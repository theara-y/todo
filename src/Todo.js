import { useState } from "react";
import "./Todo.css";
import EditTodoForm from "./EditTodoForm";

const Todo = ({ todo, idx, removeTodo, updateTodo }) => {
    const [edit, setEdit] = useState(false);

    const toggleDone = () => {
        const { text, done } = todo
        updateTodo(idx, { text, done: !done })
    };

    const toggleEdit = () => {
        setEdit(!edit);
    };

    const changeText = (newText) => {
        // toggleEdit();
        const { done } = todo;
        updateTodo(idx, { text: newText, done });
    };

    return (
        <>
            {edit ?
                <EditTodoForm changeText={changeText} currentTodo={todo.text} /> :
                <div>
                    <div
                        className={todo.done ? "Todo-done" : ""}
                        onClick={toggleDone}>{todo.text}
                    </div>
                    <button onClick={() => removeTodo(idx)}>X</button>
                    <button onClick={toggleEdit}>Edit</button>
                </div>

            }
        </>
    );
};

export default Todo;