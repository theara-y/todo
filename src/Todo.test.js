import { render } from "@testing-library/react";
import Todo from "./Todo";

const MOCK_TODO = {
    text: "test todo",
    done: false
}
it("renders", () => {
    render(<Todo todo={MOCK_TODO}/>);
});

it("matches snapshots", () => {
    const { asFragment } = render(<Todo todo={MOCK_TODO}/>)
    expect(asFragment()).toMatchSnapshot();
})