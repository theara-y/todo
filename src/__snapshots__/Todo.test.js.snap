// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<DocumentFragment>
  <div>
    <div
      class=""
    >
      test todo
    </div>
    <button>
      X
    </button>
    <button>
      Edit
    </button>
  </div>
</DocumentFragment>
`;
